lib/index.js

```diff
@@ -12,18 +12,27 @@ var pkg = require('../package.json'); // require package.json for attributes
  */
 function createConfig (config) {
   var mergedConfig = {
-    401: { message: 'Please Login to view that page' },
-    400: { message: 'Sorry, we do not have that page.' },
-    404: { message: 'Sorry, that page is not available.' }
+    templateName: 'error_template',
+    statusCodes: {
+      401: { message: 'Please Login to view that page' },
+      400: { message: 'Sorry, we do not have that page.' },
+      404: { message: 'Sorry, that page is not available.' }
+    }
   };
 
-  Object.keys(config).forEach(function (statusCode) {
-    if (!mergedConfig[statusCode]) {
-      mergedConfig[statusCode] = {};
-    }
+  // Target status code configuration objects.
+  var statusCodes = config.statusCodes || config; // Backwards compatibility.
```

Comment:
@calmdev this _could_ ensure backward-compatibility, but I think we should _consider **breaking**_ backward-compatibility to introduce this new feature... to _simplify_ the code...

However, in the interest of not _forcing_ our existing teams/projects/users to "_upgrade_" to a new major version, I _agree_ with this approach. 👍